src/parser.rs

```diff
@@ -286,17 +315,37 @@ impl<'a> Serialize for Release<'a> {
     }
 }
 
+/// Given a string checks if the release is generally valid.
+pub fn validate_release(release: &str) -> Result<(), InvalidRelease> {
```

Comment:
In order to use this function directly on a `release` attribute, we would need to ignore leading/trailing spaces. Or, better, return the trimmed string like in https://github.com/getsentry/sentry-release-parser/pull/15.

Reply:
The lack of trimming is intentional here. Whitespace is not trimmed on the backend, only in ingestion

Reply:
Ok, meaning we'll have to add the trimming in Relay on session releases. 👍